tests/hmm/hmumu_utils.py

```diff
@@ -1129,6 +1111,25 @@ def get_selected_muons(
     mu_iso_trig_matched_cut (float) - tight isolation requirement the trigger matched muon
     muon_id_trig_matched_type (string) - "tight" muon ID requirement for trigger matched muon
     """
+
+    if fsrphotons:
+        mu_attrs = ["pt", "eta", "phi", "mass", "fsrPhotonIdx"]
+        size = muons.numevents()
+
+        for i_mu in [0,1]: # FSR recovery for first two muons in an event
+            mu = muons.select_nth(i_mu, attributes=mu_attrs)
```

Comment:
just to pay attention: `select_nth` creates a copy, which makes the code slower. Once you are convinced that the physics logic is correct, we should see if there is a way to achieve the same result without copying data in memory.

Reply:
Okay, I think I have an idea how to do it. I'll commit the fix soon

Reply:
OK, great! In such cases I would always suggest to get the physics correct and then follow up with an efficient implementation.